File: C/exercise/finalPositionOfSnake.c

#include "finalPositionOfSnake.h"

int finalPositionOfSnake(int n, char** commands, int commandsSize) {
	int x = 0, y = 0;
	for (int i = 0; i < commandsSize; i++) {
		if (commands[i][0] == 'U') {
			x--;
		} else if (commands[i][0] == 'D') {
			x++;
		} else if (commands[i][0] == 'L') {
			y--;
		} else if (commands[i][0] == 'R') {
			y++;
		} else {
			// Invalid command
		}
	}
	return (x * n) + y;
}

File: Ruby/solution/final_position_of_snake.rb
# @param {Integer} n
# @param {String[]} commands
# @return {Integer}
def final_position_of_snake(n, commands)
  pos = 0
  commands.each do |cmd|
    case cmd
    when "LEFT"
      pos -= 1
    when "RIGHT"
      pos += 1
    when "DOWN"
      pos += n
    when "UP"
      pos -= n
    end
  end
  pos
end
